### PR TITLE
Update gitian-build.sh for docker

### PR DESCRIPTION
**Pull request by achow101**
Updates gitian-build.sh script for the docker virtualization method recently added to gitian-builder.

Second commit depends on https://github.com/devrandom/gitian-builder/pull/185

### CODE REVIEW

**Comment by laanwj** [comment]
Concept ACK, thanks for working on this.

File: contrib/gitian-build.sh [hunk 1]
```diff
@@ -238,6 +256,10 @@ then
     then
         sudo apt-get install lxc
         bin/make-base-vm --suite trusty --arch amd64 --lxc
+    elif [[ -n "$USE_DOCKER" ]]
+    then
+        sudo apt-get install docker-ce
+        bin/make-base-vm --suite trusty --arch amd64 --docker --docker-image-digest b8855dc848e2622653ab557d1ce2f4c34218a9380cceaa51ced85c5f3c8eb201
```
Comment (ken2812221) [inline contrib/gitian-build.sh]: Are you sure that this digest will stay the same? They update the image very often. https://hub.docker.com/r/library/ubuntu/tags/

Comment (MarcoFalke) [reply contrib/gitian-build.sh]: @achow101 Is this hash even supposed to be deterministic or does it just serve the purpose of an unique identifier?

Comment (achow101) [reply contrib/gitian-build.sh]: I'm not sure.. I think it may just serve as a unique identifier.

**Comment by fanquake** [comment]
Concept ACK, should this go in before #13623?

**Comment by achow101** [comment]
I dropped the second commit that used gitian image hashes.

**Comment by ken2812221** [comment]
Do this actually work with latest version of gitian-builder? I always get a error message `Error: No such container: gitian-target`

**Comment by achow101** [comment]
@ken2812221 It works for me. Maybe you need to run it as root (with `sudo`) or set docker to run [without root](https://docs.docker.com/install/linux/linux-postinstall/#manage-docker-as-a-non-root-user).

**Pull request merged**